srml/democracy/src/lib.rs

```diff
@@ -154,6 +155,22 @@ decl_module! {
 				runtime_io::print(e);
 			}
 		}
+
+		/// Delegate vote.
+		pub fn delegate(origin, ref_index: ReferendumIndex, to: T::AccountId, lock_periods: LockPeriods) -> Result {
+			let who = ensure_signed(origin)?;
+			<Delegations<T>>::insert(who.clone(), (to.clone(), lock_periods.clone(), ref_index));
+			Self::deposit_event(RawEvent::Delegated(who, to, ref_index));
+			Ok(())
+		}
+
+		/// Undelegate vote.
+		fn undelegate(origin, ref_index: ReferendumIndex) -> Result {
+			let who = ensure_signed(origin)?;
+			<Delegations<T>>::remove(who.clone());
+			Self::deposit_event(RawEvent::Undelegated(who, ref_index));
```

Comment:
this event might be misleading, cause currently you can also `undelegate` by `delegating` to some other `ref_index`